spec/javascripts/spec_helper.js

```diff
@@ -17,5 +17,7 @@ global.document = window.document;
 global.getComputedStyle = window.getComputedStyle;
 global.self = window;
 
+process.env.ACUANT_MINIMUM_FILE_SIZE = 0;
```

Comment:
I went back-and-forth between _a lot_ of different options for how best to mock this in the test environment:

- Use [real fixture file data](https://github.com/18F/identity-idp/tree/master/spec/fixtures/doc_auth_images), and update fixtures to passing values. Downside is it makes tests asynchronous, and also more memory-intensive in loading the files into memory, and that this needs to be handled anywhere an upload is expected, even outside the test specs for AcuantCapture (e.g. [the test for "progresses through steps"](https://github.com/18F/identity-idp/blob/476c462157ea67967bfe3bfaf66233462eff52d9/spec/javascripts/packages/document-capture/components/document-capture-spec.jsx#L26-L53)).
- Mock `getDataURLFileSize` through `proxyquire` (also began pondering [`jest.mock`](https://jestjs.io/docs/en/mock-functions#mocking-modules)). Downside is it requires that the function be defined in a separate module.
- Mock `getDataURLFileSize` through `sinon.stub`. Similar to above, this requires the code be modified in a way to support the tests, likely through something like a property on the component function (`sinon.stub( AcuantCapture, 'getDataURLFileSize' )`.
- Provide these values through [React Context](https://reactjs.org/docs/context.html) like we've done elsewhere. This could work okay, and might be nice to be able to extract these values to some configuration. But there's a decent amount of overhead involved in getting this set up.

The approach here seemed like a simple-ish option which has a benefit in removing the file size restriction across the entire test suite, while still allowing the tests which explicitly want to confirm the expected validation behavior to be able to do so (by overriding the `minimumFileSize` prop).

Reply:
the env var approach seems fine to me --- but I'd have expected something small but not zero like `1` than `0`? Otherwise the test suite just uses 500kb anyways? That way we can still check for non-empty things right?

The ternary used to detect it would need to do like a `!== undefined` check instead of the plain check it does now right? Or like maybe use `||` so that if it was `NaN` we'd have a reasonable default?

```js
const DEFAULT_ACCEPTABLE_FILE_SIZE_BYTES = Number(process.env.ACUANT_MINIMUM_FILE_SIZE) || 500 * 1024
```

Reply:
So I think it works as-is, but definitely not in a way that one would likely expect. Even though `0` is falsey and one might assume the "else" of the ternary should take effect, it doesn't actually work this way.

See:

```
$ node
Welcome to Node.js v12.18.1.
Type ".help" for more information.
> !! 0;
false
> process.env.ACUANT_MINIMUM_FILE_SIZE = 0;
0
> !! process.env.ACUANT_MINIMUM_FILE_SIZE;
true
> // Huh?!
undefined
> typeof process.env.ACUANT_MINIMUM_FILE_SIZE;
'string'
> !! '0';
true
```

Environment variables are always strings, and the string value `'0'` is truthy. So it works, but only accidentally 😄  Like you mention, `!== undefined` (or `'ACUANT_MINIMUM_FILE_SIZE' in process.env`) is probably a better alternative. The `||` wouldn't have the desired effect, since we want 0 to be the minimum in the test environment, not 500kb.

I also probably want to check that these values are being substituted correctly. I seem to recall Webpack's `process.env` handling is pretty naive (as in, replacing the literal string terms `process.env.WHATEVER`).

Reply:
And I should probably assign the value as a string, just to avoid confusion.

```suggestion
process.env.ACUANT_MINIMUM_FILE_SIZE = '0';
```

Reply:
> And I should probably assign the value as a string, just to avoid confusion.

And maybe consider enabling TypeScript checking for this directory 😄 

![image](https://user-images.githubusercontent.com/1779930/90139056-d8eb4280-dd45-11ea-996f-ddb72c10ed9a.png)

Reply:
> I also probably want to check that these values are being substituted correctly. I seem to recall Webpack's `process.env` handling is pretty naive (as in, replacing the literal string terms `process.env.WHATEVER`).

As suspected, it is naive, but it's also handled.

See:

- https://webpack.js.org/plugins/environment-plugin/
- https://github.com/rails/webpacker/blob/bc5d44d/package/environments/base.js#L31-L34

Webpacker will make all the environment variables available, and Webpack will handle the string substitution on those keys. But based on how [`DefinePlugin`](https://webpack.js.org/plugins/define-plugin/) works, they must be the actual string literal, so something like `'ACUANT_MINIMUM_FILE_SIZE' in process.env` doesn't work.